nemo_skills/code_execution/sandbox.py

```diff
@@ -292,17 +331,22 @@ def batch_evaluate_results(
         file_handles = [open(manifest, "rt", encoding="utf-8") for manifest in unroll_files(input_files)]
         cleanup_tmp_files(input_files)
 
-        def update_fn(map_to_future, line_dict):
-            line_dict["is_correct"] = map_to_future[
-                (line_dict["predicted_answer"], line_dict["expected_answer"])
-            ].result()
+        def update_fn(map_to_future, line_dict, language="python"):
+            if language == "python":
+                line_dict["is_correct"] = map_to_future[
+                    (line_dict["predicted_answer"], line_dict.get("expected_answer", ""))
+                ].result()
+            elif language == "lean4":
+                line_dict["proof_status"] = map_to_future[
```

Comment:
I think it might be easier to keep key the same for python/lean, so that you don't need to change the logic of checking for cache. Instead you can change "is_correct" to "eval_result" or something like that which is a dictionary that has "is_correct" inside which is True/False and any other keys you might need for your particular metric. If you do make this change, then also please search for all references of is_correct around the codebase and make changes accordingly

Also let's not call it "python" and "language" here, but instead call it something like answer_format, which can be "natural_language" by default or "lean" since that's what we really do here.